File: python-examples/CuriosityWeb/first_run.py
import socket

from pyspark.sql import SparkSession

spark = SparkSession \
    .builder \
    .appName("first_run") \
    .master("spark://localhost:7077") \
    .config("spark.driver.host", socket.gethostbyname(socket.gethostname())) \
    .config("spark.driver.port", "4042") \
    .getOrCreate()

input_data = [{"id": f"Id-{x + 1}", "f_name": f"FirstName-{x + 1}", "l_name": f"LastName-{x + 1}", "roll_number": x + 1}
              for x in range(20)]

df = spark.createDataFrame(input_data)

df.show()

new_df = df.where("roll_number > 5")

new_df.show()

spark.stop()

File: scala-examples/src/main/scala/CuriosityWeb/FirstRun.scala
package CuriosityWeb

import CuriosityWeb.AppConfig._
import CuriosityWeb.model.Student
import org.apache.spark.sql.SparkSession

object FirstRun extends App {

  val spark = SparkSession
    .builder()
    .appName(getClass.getSimpleName)
    .config(sparkSimpleConfig)
    .getOrCreate()

  val input = List.tabulate(20)(_ + 1).map(in => Student(s"Id-$in", s"FirstName-$in", s"LastName-$in", in))

  import spark.implicits._

  val dataset = input.toDS()

  dataset.show()

  val newDataSet = dataset.where("rollNumber > 5")

  newDataSet.show()

  spark.close()
}
